### PR TITLE
Macdougall/add gsr homepage

### PR DESCRIPTION
**Pull request by joemacd**
Updated the homepage to add GSR functionality:

Added an item to the recycler view that lists GSR reservations. Has onclick functionality to tab to My Reservations and has a "Book a Room" button.

### CODE REVIEW

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/GsrReservationsFragment.kt [hunk 1]
```diff
@@ -134,4 +134,5 @@ class GsrReservationsFragment : Fragment() {
         super.onDestroy()
         LocalBroadcastManager.getInstance(mActivity).unregisterReceiver(broadcastReceiver);
     }
+
```
Comment (joyliu-q) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/GsrReservationsFragment.kt:137]: excellent addition

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/HomeAdapter.kt [hunk 2]
```diff
@@ -470,6 +478,27 @@ class HomeAdapter(private val dataModel: HomepageDataModel) :
...
+        holder.itemView.home_gsr_subtitle?.text = "Group Study Rooms"
+        val reservations = cell.reservations
+        holder.itemView.home_gsr_rv?.layoutManager = LinearLayoutManager(mContext,
+                LinearLayoutManager.VERTICAL, false)
+        holder.itemView.home_gsr_button?.text = "Book a Room"
+        holder.itemView.home_gsr_button?.setOnClickListener {
+            mActivity.setTab(MainActivity.GSR_ID)
+            for (fragment in mActivity.supportFragmentManager.fragments) {
+                if(fragment is GsrTabbedFragment) {
+                    Log.i("GSR TEST", fragment.toString())
```
Comment (joyliu-q) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/HomeAdapter.kt:492]: hmmmmmmm log

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/HomeGsrReservationAdapter.kt [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        val toHour = to.toString("h:mm a")
+
+        val imageUrl = currentReservation.info?.get("thumbnail") ?: "https://s3.us-east-2.amazonaws.com/labs.api/dining/MBA+Cafe.jpg"
+        Picasso.get().load(imageUrl).fit().centerCrop().into(holder.itemView.item_gsr_image)
+
+        holder.itemView.item_gsr_location.text = location
+        holder.itemView.item_gsr_date.text = day + "\n" + fromHour + "-" + toHour
+
+        holder.itemView.setOnClickListener {
+            //Moves to GSR Booking Tab
```
Comment (joyliu-q) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/HomeGsrReservationAdapter.kt:67]: nit: add space after `//`

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/HomeGsrReservationAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+            mActivity.setTab(MainActivity.GSR_ID)
+
+            //Changes tab to "My Reservations" tab (from "Book a Room" tab)
```
Comment (joyliu-q) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/HomeGsrReservationAdapter.kt:70]: ditto

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/HomeGsrReservationAdapter.kt [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+            for (fragment in mActivity.supportFragmentManager.fragments) {
+                if(fragment is GsrTabbedFragment) {
+                    Log.i("GSR TEST", fragment.toString())
```
Comment (joyliu-q) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/adapters/HomeGsrReservationAdapter.kt:73]: hMMmMmmmM not again

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/GSRReservation.kt [hunk 6]
```diff
@@ -37,4 +37,21 @@ class GSRReservation {
     @Expose
     @JvmField
     var info: Map<String, String>? = null
+
+    override fun equals(other: Any?): Boolean {
+        return other is GSRReservation && this.booking_id == other.booking_id && this.name == other.name
+                && this.fromDate == other.fromDate && this.toDate == other.toDate && this.gid == other.gid
+                && this.lid == other.lid //maybe toss in info here
```
Comment (joyliu-q) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/GSRReservation.kt:44]: maybe delete comment here

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/GSRReservation.kt [hunk 7]
```diff
@@ -37,4 +37,21 @@ class GSRReservation {
     @Expose
     @JvmField
     var info: Map<String, String>? = null
+
+    override fun equals(other: Any?): Boolean {
+        return other is GSRReservation && this.booking_id == other.booking_id && this.name == other.name
```
Comment (joyliu-q) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/GSRReservation.kt:42]: why can't you just compare hashcode here? @meiron03 (Just curious feel free to roast if I'm wrong)

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/HomepageViewModel.kt [hunk 8]
```diff
@@ -24,8 +24,9 @@ import java.util.concurrent.CountDownLatch
  */
 
 class HomepageViewModel : HomepageDataModel, ViewModel() {
+    //TODO: NESTED VIEW MODEL FOR GSR BOOKINGS
```
Comment (joyliu-q) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/HomepageViewModel.kt:27]: TODO: delete comment or add space after `//`

Comment (joyliu-q) [reply PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/HomepageViewModel.kt:27]: Separate PR

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/HomepageViewModel.kt [hunk 9]
```diff
@@ -286,6 +289,25 @@ class HomepageViewModel : HomepageDataModel, ViewModel() {
...
     }
 
+    private fun getGSRReservations(studentLife: StudentLife, bearerToken: String, latch: CountDownLatch) {
+        studentLife.getGsrReservations(bearerToken).subscribe({ reservationsList ->
+            if (reservationsList.isEmpty()) {
+                addCell(HomeCell(), GSR_POS)
+
+            } else {
+                val gsrCell = GSRCell(reservationsList)
+                Log.i(TAG, "Loaded GSR Reservations")
```
Comment (joyliu-q) [inline PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/HomepageViewModel.kt:299]: Aight

**Review by joyliu-q — CHANGES_REQUESTED** [review]
Minor changes needed — someone who doesn't know Kotlin but does know good coding practices

File: PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/HomepageViewModel.kt (hunk 9 above)
Comment (joemacd) [reply PennMobile/src/main/java/com/pennapps/labs/pennmobile/classes/HomepageViewModel.kt:298]: This log statement is good methinks but ill check with Aaron

**Pull request closed without merging**